WordPress/src/main/java/org/wordpress/android/ui/mediapicker/DeviceListBuilder.kt

```diff
@@ -108,9 +107,7 @@ class DeviceListBuilder
                         getMimeType(uri),
                         dateModified
                 )
-                if (MediaUtils.isSupportedMimeType(context.contentResolver.getType(uri))) {
-                    result.add(item)
-                }
+                result.add(item)
```

Comment:
I'm not sure why we haven't done that before but we're already calling the `getMimeType(uri)` 3 lines above this code so we could be using the mime type loaded there. However, we can do that in another PR. I think the `MediaUtils.isSupportedMimeType` is quite inefficient. We can figure out something better.

Reply:
> I'm not sure why we haven't done that before but we're already calling the getMimeType(uri) 3 lines above this code so we could be using the mime type loaded there. However, we can do that in another PR. I think the MediaUtils.isSupportedMimeType is quite inefficient. We can figure out something better.

Agreed it is inefficient and as such, contributing to time consumption while building a list seems like something we don't want. 

A proposed solution is to only check in the entry points once the item has been selected, as described here https://github.com/wordpress-mobile/WordPress-Android/issues/13036

I'd say this is something to take into account and we should probably remove the `context.contentResolver.getType(uri)` call from this class altogether as well - other solutions (for example involving pre-fetching) might apply but seem potentially beyond the effort for an issue we haven't heard many times from users anyway. Let me know your thoughts @planarvoid 👍

Reply:
I was thinking about implementing pagination to solve this issue in the new media picker. I don't think it's very efficient to load all the images from the device anyway. I can imagine some users having tens of thousands of items and that could definitely cause other issues.

Reply:
Ah definitely a great idea! That passed over my head.
I was leaning on checking in the entry points given we may also need to filter out things per entry point / consumer of the picker to have a more granular control over the broad "supported by WP site" or the "give me video" or "give me images" sort of flags (although can't think of a good enough use case right now anyway, and if such a need appears it would still make more sense to filter in the picker right away)
Looking forward to seeing the new picker 💯